core/src/main/java/org/commonjava/indy/core/content/DefaultDownloadManager.java

```diff
@@ -1012,8 +1016,13 @@ private Boolean doDelete( final Transfer item, final EventMetadata eventMetadata
     {
         try
         {
-            final ConcreteResource res = item.getResource();
-            transfers.delete( res, eventMetadata );
+            Location loc = item.getLocation();
+            if ( isIgnoreReadonly( eventMetadata ) && loc instanceof CacheOnlyLocation )
+            {
+                ( (CacheOnlyLocation) loc ).setReadonly( false );
```

Comment:
I think we need to set it back to readonly after we do the deletion, if we can?

Reply:
I'll fix it on my end, I found the corresponding lines in the store method. Both leave a cached copy of CacheOnlyLocation in the HostedRepository object which is set to readonly == false with the current logic, so I'll repair that.